app/src/main/java/com/parishod/watomatic/fragment/SettingsFragment.java

```diff
@@ -0,0 +1,40 @@
+package com.parishod.watomatic.fragment;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import androidx.preference.ListPreference;
+import androidx.preference.Preference;
+import androidx.preference.PreferenceFragmentCompat;
+
+import com.parishod.watomatic.R;
+import com.parishod.watomatic.activity.main.MainActivity;
+import com.parishod.watomatic.model.preferences.PreferencesManager;
+
+public class SettingsFragment extends PreferenceFragmentCompat {
+    private ListPreference languagePref;
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        setPreferencesFromResource(R.xml.fragment_settings, rootKey);
+
+        languagePref = findPreference("pref_key_selected_app_language");
+        languagePref.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object newValue) {
+                if(!PreferencesManager.getPreferencesInstance(getActivity()).getSelectedLanguage().equals(newValue)){
+                    //switch app language here
+                    //Should restart the app for language change to take into account
+                    restartApp();
+                }
+                return true;
+            }
+        });
+    }
+
+    private void restartApp() {
+        Intent intent = new Intent(getActivity(), MainActivity.class);
```

Comment:
This should restart and open Settings activity not MainActivity. Small change but may be we can file an issue and do it later too.

Reply:
But great work @spuday90 😃👌